iroh-ctl/src/run.rs

```diff
@@ -0,0 +1,140 @@
+use std::collections::HashMap;
+use std::env;
+use std::path::PathBuf;
+
+use crate::config::{Config, CONFIG_FILE_NAME, ENV_PREFIX};
+#[cfg(feature = "fixture")]
+use crate::fixture::get_fixture_api;
+use crate::{
+    gateway::{run_command as run_gateway_command, Gateway},
+    p2p::{run_command as run_p2p_command, P2p},
+    store::{run_command as run_store_command, Store},
+};
+use anyhow::Result;
+use clap::{Parser, Subcommand};
+use iroh::{Api, Iroh};
+use iroh_metrics::config::Config as MetricsConfig;
+use iroh_resolver::resolver;
+use iroh_rpc_client::Client;
+use iroh_util::{iroh_config_path, make_config};
+
+#[derive(Parser, Debug, Clone)]
+#[clap(version, about, long_about = None, propagate_version = true)]
+pub struct Cli {
+    #[clap(long)]
+    cfg: Option<PathBuf>,
+    #[clap(long = "no-metrics")]
+    no_metrics: bool,
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+impl Cli {
+    fn make_overrides_map(&self) -> HashMap<String, String> {
+        let mut map = HashMap::new();
+        map.insert("metrics.debug".to_string(), self.no_metrics.to_string());
+        map
+    }
+}
+
+#[derive(Subcommand, Debug, Clone)]
+enum Commands {
+    // status checks the health of the different processes
+    #[clap(about = "Check the health of the different iroh processes.")]
+    Status {
+        #[clap(short, long)]
+        /// when true, updates the status table whenever a change in a process's status occurs
+        watch: bool,
+    },
+    Version,
+    P2p(P2p),
+    Store(Store),
+    Gateway(Gateway),
+    #[clap(about = "break up a file into block and provide those blocks on the ipfs network")]
+    Add {
+        path: PathBuf,
+        #[clap(long, short)]
+        recursive: bool,
+        #[clap(long, short)]
+        no_wrap: bool,
+    },
+    #[clap(
+        about = "get content based on a Content Identifier from the ipfs network, and save it "
+    )]
+    Get {
+        path: resolver::Path,
+        #[clap(long, short)]
+        output: Option<PathBuf>,
+    },
+}
+
+#[cfg(not(feature = "fixture"))]
+pub async fn run_cli(cli: Cli) -> Result<()> {
```

Comment:
these methods that take `Cli` should be implemented as 

```rust
impl Cli {
  pub async fn run(self) -> Result<() { .. }
}
```

Reply:
I've done that. It suggests further changes in p2p, store, gateway CLI handling code as well to turn functions into methods. I propose we don't do this for now as we'll be cutting away a lot of this code anyway in the near future, and I'll rearrange it then if still necessary.